scripts/ci/validate-metadata-files.js

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env node
+
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/* eslint-env node */
+
+'use strict';
+
+const CUSTOM_EXPORT_NAMES = {
+	'CKFinderUploadAdapter': 'UploadAdapter'
+};
+
+const fs = require( 'fs-extra' );
+const upath = require( 'upath' );
+const parser = require( '@babel/parser' );
+const minimist = require( 'minimist' );
+const { glob } = require( 'glob' );
+const { red, green, magenta } = require( 'chalk' );
+
+main()
+	.catch( err => {
+		console.error( err );
+
+		process.exit( 1 );
+	} );
+
+async function main() {
+	const options = getOptions( process.argv.slice( 2 ) );
+
+	console.log( magenta( '\nValidating plugins\' metadata...' ) );
+
+	const packagesGlobPattern = upath.join( options.cwd, 'packages', '*' );
+	const packagesPaths = ( await glob( packagesGlobPattern ) ).map( upath.toUnix );
+	const packagesData = await Promise.all( packagesPaths.map( getPackageData ) );
+	const packagesWithMetadata = packagesData.filter( p => p.metadata );
+
+	const invalidMetadata = [];
+
+	for ( const packageData of packagesWithMetadata ) {
+		const missingExports = getMissingExports( packageData );
+		const missingIcons = getMissingIcons( packageData, options );
+
+		if ( ![ ...missingExports, ...missingIcons ].length ) {
+			continue;
+		}
+
+		invalidMetadata.push( {
+			name: packageData.name,
+			missingExports,
+			missingIcons
+		} );
+	}
+
+	if ( !invalidMetadata.length ) {
+		console.log( green( 'Validation successful.' ) );
+
+		return;
+	}
+
+	for ( const x of invalidMetadata ) {
+		console.log( red( `\nPackage "${ x.name }" has invalid metadata entries:` ) );
+
+		if ( x.missingExports.length ) {
+			console.log( red( [
+				'Missing exports:',
+				...x.missingExports.map( missingExport => ` - ${ missingExport }` )
+			].join( '\n' ) ) );
+		}
+
+		if ( x.missingIcons.length ) {
+			console.log( red( [
+				'Missing icons:',
+				...x.missingIcons.map( missingIcon => ` - ${ missingIcon }` )
+			].join( '\n' ) ) );
+		}
+	}
+
+	process.exit( 1 );
+}
+
+/**
+ * Gets plugins from metadata file at provided path.
+ * Additionally, adds `packageName` of the plugin to returned data.
+ */
+async function getPackageData( packagePath ) {
+	const name = packagePath.split( '/' ).at( -1 );
+
+	const pkgJsonPath = upath.join( packagePath, 'package.json' );
+	const pkgJsonContent = await fs.readJSON( pkgJsonPath );
+
+	const metadataPath = upath.join( packagePath, 'ckeditor5-metadata.json' );
+	const indexPath = upath.join( packagePath, pkgJsonContent.main );
```

Comment:
If there is no `main` field in any package, then this script throws:

![obraz](https://github.com/ckeditor/ckeditor5/assets/56868128/8eceecd7-090d-4f4b-a430-932a82e073b7)

Reply:
What do you think we should do in this case?

Reply:
In this script we check the metadata of packages and if the package contains a file with metadata, it must also contain an index file to know where the plugins announced in the metadata are available.

Not all packages have an entry point and that seems OK, because apparently they don't have to have one.

So, in case a package contains a metadata file but it does not have an entry point, I would expect the script not to explode and the package to be marked as incorrect, because it has missing exports (because we do not know the index file where these exports are available).